### ISSUE TITLE
Panicking in a method will cause godot to believe the method does not exist

### ISSUE THREAD

**Issue by lilizoey** [godot-rust/gdext#254]
Using this code:
```rs
use godot::prelude::*;

#[derive(GodotClass)]
#[class(init, base=Node)]
pub struct Mini {
    #[base]
    base: Base<Node>,
}

#[godot_api]
impl Mini {
    #[func]
    fn bar(&self) {
        panic!("error")
    }
}

struct Testing;

#[gdextension]
unsafe impl ExtensionLibrary for Testing {}
```

if you attempt to call `bar` from gdscript, Godot will report that the method does not exist at runtime:
![Screenshot_20230501_021644](https://user-images.githubusercontent.com/9107253/235382939-b90813ed-411d-4f24-87e8-2d824308c14d.png)

However it will also properly report the error we print:
![Screenshot_20230501_021711](https://user-images.githubusercontent.com/9107253/235382988-e8654364-ec25-4996-8008-c3cafecb675d.png)

**Comment by Bromeon** [comment]
If Godot interprets failed functions calls as "function does not exist", is there realistically something that we can do on our side?

Should we just let the call succeed but still print an error? But then we'd need to return something fitting the signature... and this might have other negative effects?

**Comment by Bromeon** [comment]
Panics are quite readable in the meantime. When adding a panic to demo-projects/hot-reload's `get_number` method and calling from GDScript:

<img width="702" height="160" alt="Panic rendering" src="https://github.com/user-attachments/assets/76ca9d75-363a-4f0f-8cb4-b8595e20a419" />

The one thing that may be a bit confusing is the last line. That happens because of:
https://github.com/godot-rust/gdext/blob/01d3346fb9cdb56b4d8e6fe9b495b53cc26df798/godot-ffi/src/extras.rs#L99-L99

Godot doesn't like us using a non-official error type. We could consider using [one of the official ones](https://github.com/godotengine/godot/blob/9cd297b6f2a0ee660b8f1a6b385582cccf3a9d10/core/extension/gdextension_interface.h#L183-L191), but none really fits:

```c
typedef enum {
	GDEXTENSION_CALL_OK,
	GDEXTENSION_CALL_ERROR_INVALID_METHOD,
	GDEXTENSION_CALL_ERROR_INVALID_ARGUMENT, // Expected a different variant type.
	GDEXTENSION_CALL_ERROR_TOO_MANY_ARGUMENTS, // Expected lower number of arguments.
	GDEXTENSION_CALL_ERROR_TOO_FEW_ARGUMENTS, // Expected higher number of arguments.
	GDEXTENSION_CALL_ERROR_INSTANCE_IS_NULL,
	GDEXTENSION_CALL_ERROR_METHOD_NOT_CONST, // Used for const call.
} GDExtensionCallErrorType;
```
Maybe `INVALID_ARGUMENT` is the closest? Either way, _that_ specific message will only ever be as precise as the above call errors.

TLDR: panic messages are good enough now, once multiple lines are considered. Changing the call error enum or its description would require an upstream change.
